tests/rptest/tests/memory_stress_test.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2023 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from enum import Enum
+
+from ducktape.mark import ok_to_fail
+from rptest.clients.types import TopicSpec
+from rptest.services.cluster import cluster
+from rptest.services.kaf_consumer import KafConsumer
+from rptest.services.redpanda import ResourceSettings
+from rptest.services.rpk_consumer import RpkConsumer
+from rptest.services.rpk_producer import RpkProducer
+from rptest.services.verifiable_consumer import VerifiableConsumer
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.utils.mode_checks import skip_debug_mode
+
+
+class Consumers(Enum):
+    RPK = 1
+    KAF = 2
+    VERIFIABLE = 3
+
+
+class MemoryStressTest(RedpandaTest):
+    """
+    Try various ways to reach out-of-memory condition in a broker via Kafka API
+    """
+    def setUp(self):
+        # Override parent setUp so that we don't start redpanda until each test,
+        # enabling each test case to customize its ResourceSettings
+        pass
+
+    @ok_to_fail  # until the fix is delivered
+    @cluster(num_nodes=5)
+    @skip_debug_mode
+    def test_fetch_with_many_partitions(self):
+        """
+        Exhaust memory by consuming from too many partitions in a single Fetch
+        API request.
+        """
+        # memory_mb does not work with debug redpanda build, therefore the test
+        # only makes sense with release redpanda, hence @skip_debug_mode
+        self.redpanda.set_resource_settings(
+            ResourceSettings(memory_mb=256, num_cpus=1))
+        self.redpanda.set_seed_servers(self.redpanda.nodes)
+        self.redpanda.start(omit_seeds_on_idx_one=False)
+        self.redpanda.set_cluster_config(
+            {"kafka_batch_max_bytes": 10 * 1024 * 1024})
+
+        # the maximum message size that does not make redpanda OOM with all
+        # the other params as they are is 64 MiB
+        msg_size = 1024 * 1024
+        partition_count = 400
```

Comment:
This partition count may not be stable on docker tests.  This test probably needs to live in `scale_tests/` long term.

Reply:
What is the maximum I can safely use if I want it to run on docker? I would try to keep it available in docker env if possible.